orchestrator/gravity_utils/src/types/ethereum_events.rs

```diff
@@ -818,6 +759,106 @@ impl LogicCallExecutedEvent {
     }
 }
 
+fn parse_string(data: &[u8], arg_index: usize) -> Result<String, GravityError> {
+    // fetching the string from the first 32 bytes
+    let offset_start = arg_index * 32;
+    let offset_end = offset_start + 32;
+    let offset = Uint256::from_bytes_be(&data[offset_start..offset_end]).unwrap();
+
+    // parse start and end of the string length
+    let len_start_index: usize = offset.to_string().parse().unwrap();
+    let len_end_index = len_start_index + 32;
+
+    let len = Uint256::from_bytes_be(&data[len_start_index..len_end_index]).unwrap();
+    let len: usize = len.to_string().parse().unwrap();
```

Comment:
same here. The `unwrap` on `from_bytes_be` is ok because the width of the slice must be 32, but make sure that the `len_end_index` is not greater than `data.len()`.

Reply:
Thanks. Updated here and for the entire file.